### PR TITLE
fix(set cookies): set cookies failure with invalid cookie attributes

### PR DESCRIPTION
**Pull request by amiralis1365**
[issue link](https://github.com/BrowseAI-Official/BrowseAI/issues/373)

### CODE REVIEW

File: src/lib/cookies.js [hunk 1]
```diff
@@ -107,7 +107,21 @@ class CookieHandler extends CDP {
...
+                    cleanedCookie.path = decodeURIComponent(cleanedCookie.path)
+                } 
+                cookies.push(cleanedCookie);
+            }
+            return cookies;
+        }, [])
+        try {
+            await this.Network.setCookies({cookies: cleanedCookies});
+        } catch (error) {
+            throw new Error (`Could not set cookies. error: ${JSON.stringify(error)} cookies: ${JSON.stringify(cleandCookies)}`)
```
Comment (ardalann) [inline src/lib/cookies.js]: JSON.stringify works poorly with error object and their stack traces. Let's just `throw error`

File: src/lib/cookies.js [hunk 2]
```diff
@@ -107,7 +107,21 @@ class CookieHandler extends CDP {
             await this.Network.deleteCookies(badCookie);
         }
         // Store cookies in the browser
-        await this.Network.setCookies({cookies: browserCookies});
+        const cleanedCookies = browserCookies.reduce((cookies, cookie) => {
+            if (cookie.name !== "") {
```
Comment (ardalann) [inline src/lib/cookies.js]: The Hossein in my head says you should do this:
```suggestion
            if (cookie.name === "") 
              return cookies;
```

File: src/lib/cookies.js [hunk 3]
```diff
@@ -107,7 +107,21 @@ class CookieHandler extends CDP {
...
+                const cleanedCookie = cookie;
```
Comment (ardalann) [inline src/lib/cookies.js]: should be
```suggestion
                const cleanedCookie = {...cookie};
```

File: src/core/proxy.js [hunk 4]
```diff
@@ -45,6 +45,7 @@ const requestHandler = async (request, proxy, overrides = {}) => {
     } catch (error) {
         console.error("proxy requestHandler error caught:", error)
         await request.abort();
+        throw error;
```
Comment (ardalann) [inline src/core/proxy.js:48]: I don't think this gets caught anywhere. what's the point?

**Review by ardalann — CHANGES_REQUESTED** [review]

File: src/lib/cookies.js (hunk 1 above)
Comment (amiralis1365) [reply src/lib/cookies.js]: what about cookies?

File: src/core/proxy.js (hunk 4 above)
Comment (amiralis1365) [reply src/core/proxy.js:48]: it will get caught by useProxy function

**Review by ardalann — APPROVED** [review]
Looks good. We just have to make sure that the error is caught in `worker` and it doesn't make the job fail, it just gets logged to datadog as an error.

File: src/lib/cookies.js [hunk 5]
```diff
@@ -107,7 +107,21 @@ class CookieHandler extends CDP {
             await this.Network.deleteCookies(badCookie);
         }
         // Store cookies in the browser
-        await this.Network.setCookies({cookies: browserCookies});
+        const cleanedCookies = browserCookies.reduce((cookies, cookie) => {
+            if (cookie.name === "") return cookies;
+            const cleanedCookie = {...cookie};
+            if (cleanedCookie.path.includes("%2F")) {
```
Comment (mGolestan98) [inline src/lib/cookies.js]: path could be undefined
```suggestion
            if (cleanedCookie.path?.includes("%2F")) {
```

**Review by mGolestan98 — CHANGES_REQUESTED** [review]

**Pull request merged**